LLama.Unittest/SamplingTests.cs

```diff
@@ -0,0 +1,189 @@
+using LLama.Common;
+using LLama.Native;
+
+using System.Numerics.Tensors;
+using System.Runtime.InteropServices;
+using System.Text;
+
+using Xunit.Abstractions;
+
+namespace LLama.Unittest
+{
+    public class SamplingTests : IDisposable
+    {
+        private readonly ITestOutputHelper _testOutputHelper;
+        private readonly LLamaWeights _model;
+        private readonly ModelParams _params;
+
+        private readonly LLamaBatch _batch;
+        private readonly StreamingTokenDecoder _decoder;
+
+        public void Dispose() => _model.Dispose();
+
+        private unsafe Span<float> GetLogits(LLamaContext context, int totalSequences) => new(llama_get_logits(context.NativeHandle), totalSequences * _model.VocabCount);
```

Comment:
This is already available in the SafeLLamaContextHandle with a safe wrapper. See [here](https://github.com/SciSharp/LLamaSharp/blob/4436f7e28fcac6e792e6af364f899729d68ba778/LLama/Native/SafeLLamaContextHandle.cs#L441-L469).

Reply:
[Discussed in Discord](https://discord.com/channels/1106946823282761851/1106947264938790972/1306409468547956798).

tl;dr: that method isn't quite suitable. Lyrcaxis will modify it.

Reply:
Added the option to retrieve multiple logits of multiple tokens and documentation for it on the existing method:
![image](https://github.com/user-attachments/assets/dec24f6f-811b-4da4-b7a9-5c2946218181)

lmk if you feel documentation is lacking or excessive.